plugins/src/ios/OptimoveSDKPlugin.swift

```diff
@@ -1,14 +1,51 @@
-@objc(initialize:)
-  func initialize(command: CDVInvokedUrlCommand) {
-    let args = command.arguments
-    
-    let inAppConsentStrategy: InAppConsentStrategy = {
-      if let strategy = args?[2] as? String {
-        return .init(rawValue: strategy)!
-      }
-      return.notEnabled
-    }()
-
-    let config = OptimoveConfigBuilder(optimoveCredentials: args?[0] as? String, optimoveCredentials: args?[1] as? String).enableInAppMessaging(inAppConsentStrategy: inAppConsentStrategy).build()
-    Optimove.initialize(with: config)
-  }
\ No newline at end of file
+import OptimoveSDK
+import NotificationCenter
+@objc(Optimobile_Cordova) class OptimobileSDKPlugin : CDVPlugin {
+    private let internalQueue = DispatchQueue(label: "com.singletioninternal.queue",
```

Comment:
Is this used?

Reply:
yes this is for the thread safe init

Reply:
Ah yes, sorry had overlooked this.

Perhaps we could use something like the lazy property approach mentioned here https://stackoverflow.com/a/37887068 to avoid creating a queue just for initialisation? (i.e. we could de-ref some lazy prop with a block initialiser from the load method?)

Reply:
i think what i did is better

Reply:
OK, can you explain a little more why you think it's better?

My suggestion is based on the notes around excessive thread creation from https://developer.apple.com/documentation/dispatch/dispatchqueue

It seems perhaps we don't need to create a `DispatchQueue` to handle this if we use lazy properties. Because this queue would only be used for the initialisation, it seems a little wasteful.

Happy to hear your thoughts on it and comparison of approaches.

Reply:
because this is the way to do thread safety in swift
lazy is for parameters as far as i know, because there is no need to make function initialize only when first used
it is a function:)

but i will have a look at what you send
i guess you mean that the config should be a lazy global parameter that get called first time in didFinishLaunching

Reply:
make config init thread safe by making it lazy property of class